### ISSUE TITLE
Subtracting ByteTensor doesn't work in this case.

### ISSUE THREAD

**Issue by mrdrozdov** [pytorch/pytorch#3225]
This snippet used to work, but now throws an error:

```
In [1]: import torch

In [2]: torch.__version__
Out[2]: '0.2.0_4'

In [3]: from torch.autograd import Variable

In [4]: 1 - Variable(torch.cuda.ByteTensor(2).fill_(1))
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-d11efaae512d> in <module>()
----> 1 1 - Variable(torch.cuda.ByteTensor(2).fill_(1))

/home/dexter/anaconda2/lib/python2.7/site-packages/torch/autograd/variable.pyc in __rsub__(self, other)
    824
    825     def __rsub__(self, other):
--> 826         return SubConstant.apply(other, self)
    827
    828     def __mul__(self, other):

/home/dexter/anaconda2/lib/python2.7/site-packages/torch/autograd/_functions/basic_ops.pyc in forward(ctx, a, b, inplace)
    128                 return tensor.neg_().add_(constant)
    129             else:
--> 130                 return tensor.neg().add_(constant)
    131
    132     @staticmethod

AttributeError: 'torch.cuda.ByteTensor' object has no attribute 'neg'
```

**Comment by albanD** [comment]
Hi,
It is not the substraction that does not work, it is the negation.
A `ByteTensor` is an unsigned type, so doing `- byte_tensor` does not work.

**Comment by mrdrozdov** [comment]
Thanks for pointing this out. Perhaps the behavior of SubConstant should change? Here is the old behavior (0.1.12):

```
def forward(self, a):
    if self.sub_tensor:
        if a.is_signed() and self.inplace:
            self.mark_dirty(a)
            return a.neg_().add_(self.constant)
        else:
            assert not self.inplace, "can't perform (constant - tensor) " \
                "subtraction in-place on an unsigned type"
            return a.new().resize_as_(a).fill_(self.constant).sub_(a)
    else:
        if self.inplace:
            self.mark_dirty(a)
            return a.sub_(self.constant)
        else:
            return a.sub(self.constant)
```

**Comment by mrdrozdov** [comment]
It might be sufficient to update the docs as a breakage created by torch v0.2.0.

**Comment by mrdrozdov** [comment]
This was somewhat referred to in a different issue (https://github.com/pytorch/pytorch/issues/2543), but I think the problem is slightly different. There should be a simple way to invert a "binary tensor" (from 0101 -> 1010 for instance).

**Comment by colesbury** [comment]
Fixed in #4075

**Comment by mrdrozdov** [comment]
🎆
